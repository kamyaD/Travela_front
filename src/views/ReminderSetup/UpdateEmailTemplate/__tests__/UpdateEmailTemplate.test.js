import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import React from 'react';
import configureStore from 'redux-mock-store';
import UpdateEmailTemplate from '../index';

const state = {
  reminderManagement: {
    updatedEmailTemplate: {
      isSaving: false,
      errors: {},
      data: {}
    }
  }
};

const props = {
  createReminderEmailTemplate: jest.fn(),
  history: {
    push: jest.fn(),
  },
  match: {
    params: {}
  },
  editing: true
};

describe('<UpdateEmailTemplate> page', () => {
  let wrapper;

  beforeEach(() => {
    const mockStore = configureStore();
    const store = mockStore(state);
    wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <UpdateEmailTemplate {...props} editing />
        </MemoryRouter>
      </Provider>
    );
  });

  afterEach(() => {
    jest.resetAllMocks();
    wrapper.unmount();
  });

  it('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
