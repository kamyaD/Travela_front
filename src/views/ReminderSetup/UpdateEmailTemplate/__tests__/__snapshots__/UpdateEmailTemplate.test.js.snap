// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateEmailTemplate> page Should render correctly 1`] = `
<div>
  <div
    className="PageHeader"
  >
    <div>
      <a
        href="/settings/reminder-setup/"
        onClick={[Function]}
      >
        <img
          alt="icon"
          className="header__link"
          src="back-icon.svg"
        />
      </a>
      <span
        className="title"
      >
        EDIT AN EMAIL TEMPLATE
      </span>
    </div>
  </div>
  <form
    className="new-email-template-form"
    onSubmit={[Function]}
  >
    <fieldset
      className="form-fields"
    >
      <div
        className="input-group"
      >
        <div
          className="form-input  "
        >
          <label
            htmlFor="name"
          >
            Template Name
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
               * 
            </span>
          </label>
          <input
            autoComplete="off"
            className=" "
            label="Template Name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value=""
          />
          <span
            className="error"
          />
        </div>
        <div
          className="emails"
        >
          <div
            className="form-input  "
          >
            <label
              htmlFor="from"
            >
              From
              <span
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                 * 
              </span>
            </label>
            <input
              autoComplete="off"
              className=" "
              label="From"
              name="from"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value=""
            />
            <span
              className="error"
            />
          </div>
          <div
            className="form-input  "
          >
            <label
              htmlFor="cc"
            >
              Cc
            </label>
            <div
              className="ReactTags__tags react-tags-wrapper"
            >
              <div
                className="ReactTags__selected"
              >
                <div
                  className="ReactTags__tagInput"
                >
                  <input
                    aria-label="Add an email"
                    className="ReactTags__tagInputField"
                    name="cc"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onPaste={[Function]}
                    placeholder="Add an email"
                    type="text"
                  />
                </div>
              </div>
            </div>
            <span
              className="error"
            />
          </div>
        </div>
        <div
          className="form-input  "
        >
          <label
            htmlFor="subject"
          >
            Email Subject
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
               * 
            </span>
          </label>
          <input
            autoComplete="off"
            className=" "
            label="Email Subject"
            name="subject"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value=""
          />
          <span
            className="error"
          />
        </div>
        <div
          className="form-input email-message "
        >
          <label
            htmlFor="message"
          >
            Email Message
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
               * 
            </span>
          </label>
          <textarea
            autoComplete="off"
            className=" textarea-box email-message"
            label="Email Message"
            name="message"
            onBlur={[Function]}
            onChange={[Function]}
            rows="10"
            type="textarea"
            value=""
          />
          <span
            className="error"
          />
        </div>
      </div>
    </fieldset>
    <fieldset>
      <div
        className="submit-area"
      >
        <button
          className="bg-btn bg-btn--active"
          disabled={true}
          id="submit"
          type="submit"
        >
          Update
        </button>
        <button
          className="bg-btn bg-btn--inactive"
          id="cancel"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;
